### PR TITLE
Add documentation for using ComponentGraphs

### LINKED ISSUE #2367: Add documentation for using ComponentGraphs
We should add some sort of documentation for how to use component graphs. Specifically, how does a user specify a DAG-like component graph? How is compute order calculated? How should the user specify the connections and the flow of data throughout the graph?

I think some of these things are not really clear even as a developer, and it would be great to have this for us to have, too.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2367

Docs here: https://feature-labs-inc-evalml--2673.com.readthedocs.build/en/2673/user_guide/component_graphs.html

### CODE REVIEW

File: docs/source/user_guide/component_graphs.ipynb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    "cg_simple.instantiate({'My Imputer': {'numeric_impute_strategy': 'most_frequent'}})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Components in the Component Graph\n",
+    "\n",
+    "You can use `.get_component(name)` and provide the unique component name to access any component in the component graph. Below, we can grab our Imputer component and confirm that `numeric_impute_strategy` has indeed been set to \"most_frequent\"."
```
Comment (eccabay) [inline docs/source/user_guide/component_graphs.ipynb:86]: Can you add a section about `get_inputs` here as well?

File: docs/source/user_guide/component_graphs.ipynb [hunk 2]
```diff
@@ -0,0 +1,199 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Component Graphs\n",
+    "\n",
+    "EvalML component graphs represent and describe the flow of data in a collection of related components. A component graph is comprised of nodes representing components, and edges between pairs of nodes representing where the inputs and outputs of each component should go. It is the backbone of the features offered by the EvalML [pipeline](pipelines.ipynb), but is also a powerful data structure on its own. EvalML currently supports component graphs as linear and directed acyclic graphs (DAG)."
```
Comment (eccabay) [inline docs/source/user_guide/component_graphs.ipynb:9]: Maybe add a link to a description of what a DAG is here, for anyone unfamiliar with graph theory

**Review by eccabay — CHANGES_REQUESTED** [review]
This looks great! I left a couple requests for small new sections.

The biggest thing missing for me is discussion of the flow of a graph, as was mentioned in the original issue. I would love to see a section talking about compute order, and how to know what components to link together and how. There's a little bit of that implicitly shown in the larger example, but it would be great to have a whole section dedicated to discussing it.

File: docs/source/user_guide/component_graphs.ipynb [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    "\n",
+    "\n",
+    "For example, in the code example below, we have a simple component graph made up of two components: an Imputer and a Random Forest Classifer. The names used to reference these two components are given by the keys, \"My Imputer\" and \"RF Classifier\" respectively. Each value in the dictionary is a list where the first element is the component corresponding to the component name, and the remaining elements are the inputs, e.g. \"My Imputer\" represents an Imputer component which has inputs \"X\" (the original features matrix) and \"y\" (the original target).\n",
+    "\n",
+    "Each component can have a number of feature inputs (identified as \"X\" or ending in \".x\"), but can only have one target input (identified as \"y\" or ending in \".y\"). All input edges must be explicitly defined."
```
[the remark this answers is not in the document]
Comment (bchen1116) [reply docs/source/user_guide/component_graphs.ipynb:23]: Could be useful to have an example of what it looks like to have multiple X inputs

**Review by bchen1116 — APPROVED** [review]
Agreed with Becca's comments, but looks good!

File: docs/source/user_guide/component_graphs.ipynb [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    "EvalML component graphs represent and describe the flow of data in a collection of related components. A component graph is comprised of nodes representing components, and edges between pairs of nodes representing where the inputs and outputs of each component should go. It is the backbone of the features offered by the EvalML [pipeline](pipelines.ipynb), but is also a powerful data structure on its own. EvalML currently supports component graphs as linear and directed acyclic graphs (DAG)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining a Component Graph\n",
+    "\n",
+    "Component graphs can be defined by specifying the dictionary of components and edges that describe the graph. In this dictionary, each key is the name that should be used to reference a component by, and each corresponding value is a list where the first element is the component or component name, and the remaining elements are the input edges that should be connected to that component.\n",
```
[the remark this answers is not in the document]
Comment (angela97lin) [reply docs/source/user_guide/component_graphs.ipynb:18]: I love this, thank you for the context!

(Long sentences have always been my weakness, thank you for helping me break it down 😁)

**Review by chukarsten — APPROVED** [review]
Looks solid to me! Thank you!

**Pull request merged**